### PR TITLE
Expose server-side operations from SDK

### PR DESCRIPTION
**Pull request by sodic**

### CODE REVIEW

File: waspc/examples/todo-typescript/src/jobs/print.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import { PrintTimeAndNumberOfTasks } from 'wasp/server/jobs'
+import { getTotalNumberOfTasks } from 'wasp/server/operations'
+
 export const printTimeAndNumberOfTasks: PrintTimeAndNumberOfTasks<
   {},
   void
 > = async (data, context) => {
-  const count = await context.entities.Task.count()
+  const count = await getTotalNumberOfTasks(null, context)
```
Comment (sodic) [inline waspc/examples/todo-typescript/src/jobs/print.ts:8]: We should probably rework this API when the function is void.

File: waspc/examples/todo-typescript/src/setup/serverSetup.ts
Comment (sodic) [inline waspc/examples/todo-typescript/src/setup/serverSetup.ts:1]: Just formatting.

File: waspc/data/Generator/templates/sdk/server/operations/index.ts [hunk 2]
```diff
@@ -1,4 +1,13 @@
+{{={= =}=}}
 // PUBLIC API
-export * from './queries/types'
+export * from './queries/types.js'
 // PUBLIC API
-export * from './actions/types'
+export * from './actions/types.js'
```
Comment (sodic) [inline waspc/data/Generator/templates/sdk/server/operations/index.ts:5]: This became a problem when I started using the file for something other than type imports.

Comment (infomiho) [reply waspc/data/Generator/templates/sdk/server/operations/index.ts:5]: Maybe we bundle the SDK? Bundle everyting! 😂

File: waspc/examples/todo-typescript/src/jobs/print.ts (hunk 1 above)
Comment (infomiho) [reply waspc/examples/todo-typescript/src/jobs/print.ts:8]: What does this mean?

File: waspc/src/Wasp/Generator/SdkGenerator/Server/OperationsGenerator.hs [hunk 3]
```diff
@@ -18,6 +18,7 @@ import qualified Wasp.AppSpec.Query as AS.Query
 import Wasp.AppSpec.Valid (isAuthEnabled)
 import Wasp.Generator.Common (makeJsonWithEntityData)
 import Wasp.Generator.FileDraft (FileDraft)
+import Wasp.Generator.FileDraft.TemplateFileDraft (TemplateFileDraft (_tmplData))
```
Comment (infomiho) [inline waspc/src/Wasp/Generator/SdkGenerator/Server/OperationsGenerator.hs:21]: Is this import redundant maybe?

File: waspc/src/Wasp/Generator/SdkGenerator/Server/OperationsGenerator.hs [hunk 4]
```diff
@@ -41,9 +42,19 @@ genOperations spec =
...
+      genServerTs spec
     ]
 
+genServerTs :: AppSpec -> Generator FileDraft
+genServerTs spec = return $ mkTmplFdWithData relPath tmplData
+  where
+    relPath = serverOpsDirInSdkTemplatesDir </> [relfile|index.ts|]
+    tmplData =
+      object
+        [ "actions" .= map getActionData (AS.getActions spec),
```
Comment (infomiho) [inline waspc/src/Wasp/Generator/SdkGenerator/Server/OperationsGenerator.hs:54]:
```suggestion
        [ "actions" .= getActionData <$> (AS.getActions spec),
```

If you want to Haskell it up 😄

File: waspc/data/Generator/templates/sdk/server/operations/index.ts [hunk 5]
```diff
@@ -1,4 +1,13 @@
...
+{=# queries =}
+
+export { {= operationName =} } from './queries/index.js'
```
Comment (infomiho) [inline waspc/data/Generator/templates/sdk/server/operations/index.ts:8]: Is this exporting the user defined fn directly or our wrapper of their fn?

**Review by infomiho — APPROVED** [review]

**Review by Martinsos — COMMENTED** [review]
I will leave this one to @infomiho to review properly, I am just curious, does this require us to add any rules to the codemod / our excel sheet with import rewrites?

File: waspc/examples/todo-typescript/src/jobs/print.ts (hunk 1 above)
Comment (sodic) [reply waspc/examples/todo-typescript/src/jobs/print.ts:8]: It's kind of weird to ask users to send something as the first argument when the function doesn't actually need it.

Comment (infomiho) [reply waspc/examples/todo-typescript/src/jobs/print.ts:8]: Ah okay, you meant when the args are void :D 

I agree 👍

**Pull request merged**